snapcraft/commands/upload.py

```diff
@@ -0,0 +1,64 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+snapcraft upload
+
+Upload snap package to the Ubuntu Store.
+
+Usage:
+  upload [options]
+
+Options:
+  -h --help             show this help message and exit.
+
+"""
+
+import logging
+import os
+import sys
+
+from click_toolbelt.api import upload
+
+from snapcraft import lifecycle
+
+
+logger = logging.getLogger(__name__)
+
+
+def _format_snap_name(snap):
+    snap['arch'] = snap['arch'][0] if len(snap['arch']) == 1 else 'multi'
+    return '{name}_{version}_{arch}.snap'.format(**snap)
+
+
+def main(argv=None):
+    """Upload snap package to the Ubuntu Store."""
+
+    # make sure the full lifecycle is executed
+    snap = lifecycle.execute('strip')
```

Comment:
you probably want to run the `snap` command here. This produces no snap.

Maybe some code reorg is needed if the snap command becomes common and just calling snap.main() is to cumbersome (or ugly).